From 397d329b600cfb642315896ba94aa44991f9eee7 Mon Sep 17 00:00:00 2001
From: Angela Czubak <aczubak@marvell.com>
Date: Thu, 23 May 2019 16:45:26 +0200
Subject: [PATCH 254/386] octeontx: fix pko_pf_create_domain

It is possible that domain won't contain any BGX or SDP interfaces, but
will contain LBK ones. This commit solves the issue where only one LBK
is requested but instead we calculate BGX MAC.

Change-Id: I4aac591258ed0c5d14b93477307cf099b9ceb056
Signed-off-by: Angela Czubak <aczubak@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/9828
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
index ca867fab3030..2cadcb64316c 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
@@ -370,7 +370,7 @@ static int pko_pf_create_domain(u32 id, u16 domain_id, u32 pko_vf_count,
 	struct pkopf *curr;
 	struct pci_dev *virtfn;
 	resource_size_t vf_start;
-	int i, pko_mac = sdp_count ? PKO_MAC_HOST : PKO_MAC_BGX;
+	int i, pko_mac;
 	int vf_idx = 0, port_idx = 0;
 	int mac_num, mac_mode, chan, ret = 0;
 	const u32 max_frame = 0xffff;
@@ -378,6 +378,13 @@ static int pko_pf_create_domain(u32 id, u16 domain_id, u32 pko_vf_count,
 	if (!kobj)
 		return -EINVAL;
 
+	if (sdp_count)
+		pko_mac = PKO_MAC_HOST;
+	else if (bgx_count)
+		pko_mac = PKO_MAC_BGX;
+	else
+		pko_mac = PKO_MAC_LBK;
+
 	mutex_lock(&octeontx_pko_devices_lock);
 	list_for_each_entry(curr, &octeontx_pko_devices, list) {
 		if (curr->id == id) {
-- 
2.17.1

